data/scripts.js

```diff
@@ -604,8 +604,10 @@ let BattleScripts = {
 		/** @type {(Pokemon | false | null)[]} */
 		let targetsCopy = targets.slice(0);
 		let i;
-		for (i = 0; i < hits && !damage.includes(false); i++) {
+		for (i = 0; i < hits; i++) {
```

Comment:
If you put the `pokemon.hp` check here then this will happen after the `i++` automatically.

Reply:
I would rename `i` to `hit` and `hits` to `hitCount`.

Reply:
I would also probably do `hit = 1; hit <= hitCount; hit++`

So `hit` matches `move.hit`.

Reply:
The hp check is at the end so that explosion still does damage, but I hadn’t thought of just further checking whether or not we’re on the first hit. On that note, @DaWoblefet how does Parental Bond Final Gambit work?

Reply:
I very much don't want the `pokemon.hp` check here.

Reply:
Parental Bond Final Gambit only hits once.

Reply:
Ugh, Explosion. Thanks for explaining. I still think the extra `++` is awkward though, but maybe not as awkward as switching to e.g.

    let hit = 1;
    do {
      // ...
    } while (++hit <= targetHits && pokemon.hp && !damage.includes(false));